File: framework/src/test/java/org/synsystems/onlypass/framework/serialisation/TestDateTimeSerialiser.java
package org.synsystems.onlypass.framework.serialisation;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.joda.time.DateTime;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class TestDateTimeSerialiser {
  private Gson gson;

  @Before
  public void setup() {
    // Serialiser is not used directly, so instead pass an instance to Gson
    gson = new GsonBuilder()
        .registerTypeAdapter(DateTime.class, new DateTimeSerialiser())
        .create();
  }

  @Test
  public void testSerialiseAndDeserialiseWithGson() {
    final DateTime date = new DateTime(2018, 12, 25, 12, 30);

    final String serialisedDate = gson.toJson(date);
    final DateTime deserialisedDate = gson.fromJson(serialisedDate, DateTime.class);

    assertThat(date, is(deserialisedDate));
  }
}

File: framework/src/test/java/org/synsystems/onlypass/framework/serialisation/TestDateTimeSerialiser.kt
package org.synsystems.onlypass.framework.serialisation

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.MatcherAssert.assertThat
import org.joda.time.DateTime
import org.junit.Before
import org.junit.Test

class TestDateTimeSerialiser {

  private lateinit var gson: Gson

  @Before
  fun setup() {
    gson = GsonBuilder()
        .registerTypeAdapter(DateTime::class.java, DateTimeSerialiser())
        .create()
  }

  @Test
  fun testSerialiseAndDeserialiseWithGson() {
    val date = DateTime(2018, 12, 25, 12, 30)

    val serialisedDate = gson.toJson(date)
    val deserialisedDate = gson.fromJson(serialisedDate, DateTime::class.java)

    assertThat(date, `is`(deserialisedDate))
  }
}